builder/src/Command/BuildCommand.php

```diff
@@ -0,0 +1,309 @@
+                            if (!isset($translations[$locale])) {
+                                $translations[$locale] = [];
+                            }
+                            $t = [
+                                'message' => $translation,
+                                'description' => 'Sidebar title for directory ' . $file->getPathname(),
+                            ];
+                            $translations[$locale]['sidebar.techdocSidebar.category.'.$file->getFilename()] = $t;
+                            $translations[$locale]['sidebar.userdocSidebar.category.'.$file->getFilename()] = $t;
+                        }
+                    }
+
+                    $scan($subDir . '/' . $file->getFilename(), $depth + 1);
+                }
+            }
+        };
+
+        $output->writeln(sprintf("Compiling %s to %s", realpath($dir), realpath($projectDir)));
+        $scan('');
+
+        // Dump version
+        $target = $projectDir . '/version.json';
+        $version = [
+            'refname' => getenv('PHRASEA_REFNAME'),
+            'reftype' => getenv('PHRASEA_REFTYPE'),
+            'datetime' => getenv('PHRASEA_DATETIME'),
+        ];
+        file_put_contents($target, json_encode($version, JSON_PRETTY_PRINT | JSON_UNESCAPED_SLASHES | JSON_UNESCAPED_UNICODE));
+        $output->writeln("Wrote version to: " . realpath($target));
+
+        // Dump translations to json files
+        foreach ($translations as $locale => $translation) {
+            $target = $projectDir . '/i18n/' . $locale . '/docusaurus-plugin-content-docs/current.json';
+            if (!file_exists(dirname($target))) {
+                $this->filesystem->mkdir(dirname($target));
+            }
+            file_put_contents($target, json_encode($translation, JSON_PRETTY_PRINT | JSON_UNESCAPED_SLASHES | JSON_UNESCAPED_UNICODE));
+            $output->writeln('Wrote translations to: ' . realpath($target));
+        }
+
+        $apps = ['databox', 'expose', 'uploader'];
+
+        // Create the API documentation from the JSON schema
+        foreach ($apps as $app) {
+            $this->runCommand(
+                ['pnpm', 'run', 'gen-api-docs', $app],
+                $projectDir,
+                $output
+            );
+
+            // ========== fix for api auto-generated sidebar (https://github.com/facebook/docusaurus/discussions/11458)
+            //            we add a "key" to each item
+
+            $docDir = $projectDir . '/docs/' . $app . '_api';
+            $output->writeln("Patching sidebar.ts to add keys.");
+            $this->filesystem->copy(
+                $docDir . '/sidebar.ts',
+                $docDir . '/sidebar-bkp.ts',
+                true
+            );
+            $this->filesystem->dumpFile(
+                $docDir . '/sidebar.ts',
+                preg_replace(
+                    "/(( *)id: (.*),)/m",
+                    "$1\n$2key: $3,",
```

Comment:
to avoid duplicated keys, include the current `$app`: `"$1\n$2key: " . $app . "_$3,",`